rules/Transform/Rector/MethodCall/MethodCallToMethodCallRector.php

```diff
@@ -147,23 +144,10 @@ private function isMethodCallCurrentClass(MethodCall $methodCall): bool
 
     private function isMatch(
         MethodCall $methodCall,
-        MethodCallToMethodCall $methodCallToMethodCall,
-        bool $isMethodCallCurrentClass,
-        Class_ $class
+        MethodCallToMethodCall $methodCallToMethodCall
     ): bool {
         $oldTypeObject = new ObjectType($methodCallToMethodCall->getOldType());
 
-        if ($isMethodCallCurrentClass) {
-            $className = (string) $this->nodeNameResolver->getName($class);
-            $objectType = new ObjectType($className);
-
-            if (! $objectType->equals($oldTypeObject)) {
```

Comment:
On `$this` usage, object equality is needed.

Reply:
Hey,
I don't understand why it's needed. The tests where a method from the current class is modified are still passing.
Do you have an idea of a test case that would help to see a potential issue?

Reply:
that possibly can happen on child class, the original purpose of this rule as far as I understand is to change Method Call from object passed from Property, so we should avoid happen on child class, only current class (same class).

Reply:
Oh.
Indeed, we need to clarify this. I understood it the way around. Handling the case of a method from the current class seemed to go in the direction of replacing all method calls, even from the class hierarchy.

Maybe having two rules would be a better idea?

Reply:
Another possible solution is make the rule configurable, eg: `$allowChangeChildClass`, you can implements `Rector\Core\Contract\Rector\AllowEmptyConfigurableRectorInterface` for it, you can take a look at `TypedPropertyRector` for example:

https://github.com/rectorphp/rector-src/blob/005ca33bcbc23aa83608e0e4c01b16f745a35eea/rules/Php74/Rector/Property/TypedPropertyRector.php#L42

https://github.com/rectorphp/rector-src/blob/005ca33bcbc23aa83608e0e4c01b16f745a35eea/rules/Php74/Rector/Property/TypedPropertyRector.php#L44-L57

https://github.com/rectorphp/rector-src/blob/005ca33bcbc23aa83608e0e4c01b16f745a35eea/rules/Php74/Rector/Property/TypedPropertyRector.php#L71-L74

with default to false, so you can change child class `$this->allowChangeChildClass` set to true.

Reply:
The rule is already configurable. Do you think it would be better to configure allowChangeChildClass globally or for each `MethodCallToMethodCall` ?

I'm more in favor of the second option, as one might want to have different behavior for each replacement.

And, how should it work for a replacement of a method from parent?

Reply:
I prefer to not break existing behaviour, and improve the configurable `MethodCallToMethodCall` in different PR so it will ease to review.